### PR TITLE
Added person, student and teacher classes.

### PR DESCRIPTION
**Pull request by Lockless404**
Created person, student, and teacher classes, with specific variables and methods.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,9 +1,14 @@
 # School Library
```
Comment (SajjadAhmad14) [inline README.md:2]: Please add instructions to run the app locally. It is important to add installation instructions for the user who is not a programmer.

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
+class Person
+  def initialize(age, name = 'unknown', parent_permission: true)
+    @id = Random.rand(1..1000)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+
+  attr_reader :id
+  attr_accessor :name, :age
```
Comment (SajjadAhmad14) [inline person.rb]: Please add these getters and setters before the constructor and right after the class definition starts as it's the Ruby class convention.

File: person.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+  def of_age?
+    @age >= 18
+  end
+
+  def can_use_service?
+    is_of_age? == true || @parent_permission
```
Comment (SajjadAhmad14) [inline person.rb]: Please use the `#of_age?` method here to check the condition of the person age.

File: student.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+require './person'
```
Comment (SajjadAhmad14) [inline student.rb]: Please use `require_relative` here to load a local file as `require` is used to load installed gems.

File: teacher.rb [hunk 5]
```diff
@@ -0,0 +1,12 @@
+require './person'
```
Comment (SajjadAhmad14) [inline teacher.rb]: Please use `require_relative` here to load a local file as `require` is used to load installed gems.

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi Hamza, 
 ### You have been doing a tremendous job so far :ok_hand:

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
Git flow is followed. :heavy_check_mark:
## To Improve 🙏

- [ ] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
**_Note: Please do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by Emmyn5600 — APPROVED** [review]
Hi @Lockless404 

STATUS: APPROVED :partying_face: 

Wooow👏, Your project is complete!  There is nothing else to say other than... it's time to merge it. 👏

Congratulations! 🎉 

Optional suggestions

-N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**